compiler/rustc_data_structures/src/sync.rs

```diff
@@ -31,37 +31,38 @@ pub use vec::AppendOnlyVec;
 
 mod vec;
 
-mod mode {
+mod thread_mode {
     use super::Ordering;
     use std::sync::atomic::AtomicU8;
 
     const UNINITIALIZED: u8 = 0;
-    const INACTIVE: u8 = 1;
-    const ACTIVE: u8 = 2;
+    const SINGLE: u8 = 1;
+    const MULTI: u8 = 2;
 
-    static MODE: AtomicU8 = AtomicU8::new(UNINITIALIZED);
+    static THREAD_MODE: AtomicU8 = AtomicU8::new(UNINITIALIZED);
 
+    // Weather control thread safety dynamically
     #[inline]
-    pub fn active() -> bool {
-        match MODE.load(Ordering::Relaxed) {
-            INACTIVE => false,
-            ACTIVE => true,
+    pub fn is_dyn_thread_safe() -> bool {
+        match THREAD_MODE.load(Ordering::Relaxed) {
+            SINGLE => false,
+            MULTI => true,
             _ => panic!("uninitialized parallel mode!"),
         }
     }
 
     // Only set by the `-Z threads` compile option
-    pub fn set(parallel: bool) {
-        let set: u8 = if parallel { ACTIVE } else { INACTIVE };
+    pub fn set_thread_mode(parallel: bool) {
```

Comment:
I guess this should be called `set_dyn_thread_mode` now. `SINGLE` -> `DYN_NOT_THREAD_SAFE`, `MULTI` -> `DYN_THREAD_SAFE`. `THREAD_MODE` -> `DYN_THREAD_MODE`. `mod thread_mode` -> `mod dyn_thread_mode`, or maybe keep that as just `mode` as the functions mentions `dyn` already.

Reply:
Hm..  thread mode sounds a bit odd to me. Maybe `set_dyn_thread_safety_mode` is better? or `set_dyn_sync_mode`.

Reply:
Synchronization is for thread safety. So I think we use `dyn_thread_safe` in the exposed function and `dyn_sync` inside the `sync` module